src/vs/platform/quickinput/common/quickInput.ts

```diff
@@ -28,6 +28,7 @@ export interface IQuickPickItem {
 	type?: 'item';
 	id?: string;
 	label: string;
+	filterStartIndex?: number;
```

Comment:
I don't think this is a great way to tackle the problem because this won't work when a label is 'Foo $(icon) Bar'. Wouldn't it be better to simply remove/ignore icons from filtering? E.g you could parse them out before filtering and adjust the matches accordingly. If I am not mistaken we already do this in some places. @bpasero might know where/how to do this

Reply:
There is:

https://github.com/microsoft/vscode/blob/f14e53920e3ccb2721eb4ce018c54d1b212b5972/src/vs/base/common/iconLabels.ts#L30-L36

Reply:
After a quick look, it seems like `matchesFuzzyIconAware(picker.value, parseLabelWithIcons(item.label));` is close as it provides matches excluding icons while leaving the string alone but doesn't provide any scoring for how to sort it.

Reply:
...thats actually the one I was trying to find, yeah!